pkg/kv/kvserver/raftlog/entry.go

```diff
@@ -28,6 +28,69 @@ import (
 	"go.etcd.io/etcd/raft/v3/raftpb"
 )
 
+// EntryData reflects the information returned by DecomposeEntryData.
+type EntryData struct {
+	Type raftpb.EntryType
+
+	// ConfChangeBytes is a marshaled raftpb.ConfChange{,V2}.
+	// Non-nil if and only if Type is EntryConfChange{,V2}
+	ConfChangeBytes []byte // raftpb.ConfChange{,V2}
```

Comment:
I don't see a clear benefit of passing this through if it's not parsed. Maybe limit this decomposition func to only `Normal` entries? E.g.:

```golang
func DecomposeCommand(e *raftpb.Entry) (version, bytes, etc, ok bool) {
	if e.Type != raftpb.EntryNormal {
		return ..., false
	}
	return ..., true
}
```

Reply:
Or assert?